Travel Scheduler/View Controllers/Place.m

```diff
@@ -101,4 +118,133 @@ - (void)getListOfPlacesCloseToPlaceWithName:(NSString *)centerPlaceName withComp
+        self.specificType = @"restaurant";
+    }
+    else if([self.types containsObject:@"lodging"]) {
+        self.specificType = @"hotel";
+    }
+    else {
+        self.specificType = @"attraction";
+    }
+}
+
+#pragma mark - methods to make the dictionary of opening times
+- (void)makeDictionaryOfOpeningTimes{
+    for(int dayIndexInt = 0; dayIndexInt <= 6; ++dayIndexInt){
+        NSNumber *dayIndexNSNumber = [[NSNumber alloc] initWithInt:dayIndexInt];
+        [self formatTimeForDay:dayIndexNSNumber];
+    }
+}
+
+-(void)formatTimeForDay:(NSNumber *)day {
+    int dayInt = [day intValue];
+    NSDictionary *dayDictionary = self.unformattedTimes[@"periods"][dayInt];
+    float openingTimeFloat;
+    float closingTimeFloat;
+    
+    if([dayDictionary objectForKey:@"open"] == nil) {
+        //Always closed
+        openingTimeFloat = -1;
+        closingTimeFloat = -1;
+    }
+    else if([dayDictionary objectForKey:@"close"] == nil) {
+        //Always open
+        openingTimeFloat = 0;
+        closingTimeFloat = 0;
+    }
+    else {
+        NSString *closingTimeString = dayDictionary[@"close"][@"time"];
+        closingTimeFloat = [self getTimeFromString:closingTimeString];
+        NSString *openingTimeString = dayDictionary[@"open"][@"time"];
+        openingTimeFloat = [self getTimeFromString:openingTimeString];
+    }
+    
+    NSNumber *openingTimeNSNumber = [[NSNumber alloc] initWithFloat:openingTimeFloat];
+    NSNumber *closingTimeNSNumber = [[NSNumber alloc] initWithFloat:closingTimeFloat];
+    
+    NSMutableDictionary *newDictionaryForDay = [[NSMutableDictionary alloc] init];
+    newDictionaryForDay[@"opening"] = openingTimeNSNumber;
+    newDictionaryForDay[@"closing"] = closingTimeNSNumber;
+    if([self.specificType isEqualToString:@"restaurant"]) {
+        newDictionaryForDay[@"periods"] = [self getRestaurantsPeriodsArrayFromOpeningTime:openingTimeFloat toClosingTime:closingTimeFloat];
+    }
+    else {
+        newDictionaryForDay[@"periods"] = [self getAttractionsPeriodsArrayFromOpeningTime:openingTimeFloat toClosingTime:closingTimeFloat];
+    }
+    //NSString *dayString = [NSString stringWithFormat:@"%@",day];
+    self.openingTimesDictionary[day] = newDictionaryForDay;
+    
+}
+
+-(float)getTimeFromString:(NSString *)timeString{
+    NSString *hourString = [timeString substringToIndex:2];
+    int hourInt = [hourString intValue];
+    NSString *minuteString = [timeString substringFromIndex:2];
+    float minuteIntRaw = [minuteString floatValue];
+    float minuteInt = minuteIntRaw/60;
+    float formattedTime = hourInt + minuteInt;
+    return formattedTime;
+}
+
+-(NSMutableArray *)getAttractionsPeriodsArrayFromOpeningTime:(float)openingTime toClosingTime:(float)closingTime {
+    NSMutableArray *arrayOfPeriods = [[NSMutableArray alloc] init];
+    
+    if (openingTime < 0) {
+        //Closed all day
+        return arrayOfPeriods;
+    }
+    
+    if (openingTime == 0 && closingTime == 0){
```

Comment:
I'm not sure if this is an issue but sometimes when you compare floats with ints, the float might be 0.0000 and the int is 0 and the program might say they're unequal. Not sure if this is an actual problem in this case, but it might be safe to make everything an int or make everything an nsnumber when comparing?

Reply:
Good point, thanks! Just fixed it with fabsf(actualValue - expectedValue) < 0.1